### PR TITLE
Add tests and remove empty-struct defaults that were previous ignored.

### LINKED ISSUE #2835: Testing: avoid zero port number in DB connection
We had a zero value for events.sql.connection.dbport cause a crash loop on startup for Minder.  Add some explicit tests that defaults and config merging operate as expected.

### PR DESCRIPTION
**Pull request by evankanderson**
# Summary

See https://github.com/stacklok/minder/pull/2822, which was an emergency revert to unblock Minder rollouts.  I'm going to work on a cleaner approach.

Fixes #2835

## Change Type

- [x] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [x] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

Added go tests; validated that the code pre-#2822 would fail the test, and the code pre-#2633 and post this PR both passed the test.

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [x] Checked that related changes are merged.

### CODE REVIEW

**Comment by jhrozek** [comment]
If you fix the lint warning I'll ack right away.
Thank you for working on this!

File: internal/config/server/config.go [hunk 1]
```diff
@@ -91,11 +91,21 @@ func setViperStructDefaults(v *viper.Viper, prefix string, s any) {
 
 		if field.Type.Kind() == reflect.Struct {
 			setViperStructDefaults(v, valueName+".", reflect.Zero(field.Type).Interface())
+			// TODO: we could allow `default` tags on structs to override the defaults inside the type.
```
[the remark this answers is not in the document]
Comment (evankanderson) [reply internal/config/server/config.go:94]: I think @Vyom-Yadav has a valid point for the re-use of DatabaseConfig inside the EventsConfig structure

**Review by jhrozek — APPROVED** [review]
LGTM

**Review by teodor-yanev — APPROVED** [review]
